src/fdosecrets/widgets/SettingsWidgetFdoSecrets.cpp

```diff
@@ -263,6 +267,15 @@ SettingsWidgetFdoSecrets::SettingsWidgetFdoSecrets(FdoSecretsPlugin* plugin, QWi
 
     m_ui->tabWidget->setEnabled(m_ui->enableFdoSecretService->isChecked());
     connect(m_ui->enableFdoSecretService, &QCheckBox::toggled, m_ui->tabWidget, &QTabWidget::setEnabled);
+
+    // background checking
+    m_checkTimer.setInterval(2000);
+    connect(&m_checkTimer, &QTimer::timeout, this, &SettingsWidgetFdoSecrets::checkDBusName);
+    connect(m_plugin, &FdoSecretsPlugin::secretServiceStarted, &m_checkTimer, &QTimer::stop);
+    connect(m_plugin,
+            &FdoSecretsPlugin::secretServiceStopped,
+            &m_checkTimer,
+            static_cast<void (QTimer::*)()>(&QTimer::start));
```

Comment:
Just noticed this weird cast here. What's that for?

Reply:
`QTimer::start` has multiple overloads. With the new connect syntax, you have to use `qOverload` (c++14) or `QOverload` (c++11) to select the appropriate one ([doc](https://doc.qt.io/qt-5/signalsandslots-syntaxes.html#selecting-overloaded-signals-and-slots)). But those are only available in Qt 5.7+.

Reply:
Ah, right. This is extremely ugly and I would almost say use the old SIGNAL/SLOT syntax. What do you think @droidmonkey?

Reply:
Yes in these cases I prefer the old style syntax. When implicit casts are not possible Qt will automatically find the correct connection.